caldav/server.go

```diff
@@ -42,6 +40,11 @@ type Backend interface {
 	webdav.UserPrincipalBackend
 }
 
+type MultiCalendarBackend interface {
+	Calendars(ctx context.Context) ([]*Calendar, error)
+	ListCalendarObjectsForCalendar(ctx context.Context, cal *Calendar, req *CalendarCompRequest) ([]CalendarObject, error)
```

Comment:
I went with this take a `*Calendar` now, but in some cases similar methods take a `path string` instead, I'm not sure which one we should go for?